tools/generate_plugin_artifacts.py

```
@@ -222,13 +236,19 @@ def generate_schemas(check: bool = False) -> int:
                 ).model_json_schema()
-                plug_schema.setdefault("$defs", {}).update(meta_schema.pop("$defs", {}))
-                entry["properties"]["metadata"] = meta_schema
+                plug_schema["$defs"].update(meta_schema.pop("$defs", {}))
+                # Add metadata override to the allOf
+                entry["allOf"].append(
+                    {
+                        "properties": {"metadata": meta_schema},
+                        "required": ["metadata"],
+                    }
+                )
             except Exception as e:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)